File: frontend/src/api/mcp.ts
import type { MCPServerConfig } from '../types/mcp';

const API_BASE = '/api';

export async function listMCPServers(): Promise<MCPServerConfig> {
  const response = await fetch(`${API_BASE}/mcp/servers`);
  if (!response.ok) {
    throw new Error('Failed to list MCP servers');
  }
  return response.json();
}

export async function refreshMCPServers(): Promise<MCPServerConfig> {
  const response = await fetch(`${API_BASE}/mcp/servers/refresh`, {
    method: 'POST',
  });
  if (!response.ok) {
    throw new Error('Failed to refresh MCP servers');
  }
  return response.json();
}

File: src/copilot_console/app/routers/mcp.py
"""MCP servers router - list available MCP servers and manage session selections."""

from fastapi import APIRouter

from copilot_console.app.models.mcp import MCPServer, MCPServerConfig
from copilot_console.app.services.mcp_service import mcp_service
from copilot_console.app.services.logging_service import get_logger

logger = get_logger(__name__)

router = APIRouter(prefix="/mcp", tags=["mcp"])


@router.get("/servers", response_model=MCPServerConfig)
async def list_mcp_servers() -> MCPServerConfig:
    """Get all available MCP server configurations.
    
    Returns servers from:
    - Global config: ~/.copilot/mcp-config.json
    - Plugin configs: ~/.copilot/installed-plugins/copilot-plugins/[plugin]/.mcp.json
    """
    return mcp_service.get_available_servers()


@router.post("/servers/refresh", response_model=MCPServerConfig)
async def refresh_mcp_servers() -> MCPServerConfig:
    """Force refresh the MCP server cache from disk."""
    return mcp_service.refresh()
